### PR TITLE
Do not escape select HTML entities

### PR DESCRIPTION
**Pull request by westy92**
Fixes https://github.com/flutter/flutter_markdown/issues/42.

The [underlying library](https://github.com/dart-lang/markdown) was converting `&` to `&amp;` and `<` to `&lt;`. After digging through their documentation, I couldn't find a clean way to prevent this, so I looked at the [source](https://github.com/dart-lang/markdown/blob/master/lib/src/inline_parser.dart#L30-L34) and came up with this solution. It seems a bit hacky, but works.

An ideal solution may instead be an upstream option.

### CODE REVIEW

**Comment by Hixie** [comment]
I don't understand the underlying bug here. Can you provide some examples of inputs and what we get out of the underlying library?

**Comment by westy92** [comment]
```dart
new MarkdownBody(data: 'I <3 Dart & Flutter');
```
shows up as `I &lt;3 Dart &amp; Flutter`

**Comment by Hixie** [comment]
How is the underlying library telling us that, though? According to https://github.com/dart-lang/markdown/issues/211#issuecomment-379431739, the output doesn't contain `&copy;`, it contains an actual copyright symbol. Does the output for `<` contain `&lt;`? That seems weird. Who is escaping the output?

**Comment by westy92** [comment]
As far as I can tell, the underlying library is just escaping `<` and `&` characters: https://github.com/dart-lang/markdown/blob/master/lib/src/inline_parser.dart#L28-L34

File: .gitignore [hunk 1]
```diff
@@ -3,6 +3,7 @@
 .idea
 .packages
 .pub/
+ios/
```
[the remark this answers is not in the document]
Comment (westy92) [reply .gitignore]: Running `flutter test` generates an `ios` folder for me which contains a file that says `// This is a generated file; do not edit or check into version control.`

**Comment by Hixie** [comment]
Oh wow, ok.

**Comment by Hixie** [comment]
@westy92 Will you be able to continue working on this? It's totally fine if not; we can take it over (though it might take longer). I'm just asking because I'm trying to clean out our review queue.

**Comment by Hixie** [comment]
An upstream solution would definitely be preferred, from my perspective. Maybe try approaching the maintainers of that package and see if they're interested? If not, we can always work around it here as per this PR.

**Comment by Hixie** [comment]
Let me know if you need any help getting in touch with the relevant folks.

**Comment by westy92** [comment]
I created an upstream PR: https://github.com/dart-lang/markdown/pull/218.

**Comment by westy92** [comment]
Upstream PR has been merged. :tada: 
I will update this PR once there is a new release of the upstream package.

File: test/flutter_markdown_test.dart [hunk 2]
```diff
@@ -337,13 +337,13 @@ void main() {
     }
   });
 
-  testWidgets('Less than', (WidgetTester tester) async {
-    final String mdLine = 'Line 1 <\n\nc < c c\n\n< Line 2';
+  testWidgets('HTML entities', (WidgetTester tester) async {
+    final String mdLine = 'Line 1 <\n\nc < > c c\n\n< Line & © 2';
```
[the remark this answers is not in the document]
Comment (westy92) [reply test/flutter_markdown_test.dart]: > In regular inline text, `&amp;` should turn into a regular ampersand, but in `code` blocks it should be left alone.

This is currently not the case (nor the initial intention of this PR). Should I try to add code that achieves this functionality?

**Comment by westy92** [comment]
Is there anything else that is needed to move this along?

File: test/flutter_markdown_test.dart [hunk 3]
```diff
@@ -337,13 +337,21 @@ void main() {
...
-    _expectTextStrings(
-        widgets, <String>['Line 1 &lt;', 'c &lt; c c', '&lt; Line 2']);
+    testWidgets('< to &lt; when parsing', (WidgetTester tester) async {
+      await tester.pumpWidget(_boilerplate(const Markdown(data: '<')));
+      _expectTextStrings(tester.allWidgets, <String>['<']);
+    });
+
+    testWidgets('existing HTML entities when parsing', (WidgetTester tester) async {
+      await tester.pumpWidget(_boilerplate(const Markdown(data: '&amp; &copy; &#60; &#x0007B;')));
+      _expectTextStrings(tester.allWidgets, <String>['&amp; &copy; &#60; &#x0007B;']);
```
Comment (Hixie) [inline test/flutter_markdown_test.dart]: Is this what we want? I would have expected HTML character references to get decoded by the Markdown parser, if we're saying that that's how you escape a character in Markdown.

**Comment by Hixie** [comment]
Let me know if you want to do anything about the character reference issue. I'm going to guess that's probably just an upstream issue. If you're ok with it as is, let me know, and I'll land it. Thanks!

**Comment by westy92** [comment]
I'm okay with this for now, thanks!

**Comment by Hixie** [comment]
Thanks for your contribution!

**Pull request merged**